runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -402,6 +402,17 @@ private void translateS3AndKinesisInputs(
       S3Config s3Config = getS3Config(userS3Config);
       List<EventConfig> eventConfigs = getEventConfigs(events);
 
+      // in case of default, [kinesis_parallelism = job_parallelism - s3.parallelism]
```

Comment:
Could you explain the logic behind this? What if I set don't set a kinesis parallelism and set my s3 parallelism to be the same as my job parallelism (maybe accidentally)?

Reply:
Fair point! Apparently, my code is addressing the case when `parallelism` isn't provided to  `s3_and_kinesis`  (which currently is the case on Kinesis and Kafka PTransforms in Beam) and we do some math behind. 

However, if a user doesn't provide valid and/or sufficient parallelism to S3 and Kinesis, the job will fail (with or without my code).